src/node_crypto.cc

```diff
@@ -4280,8 +4280,14 @@ void CipherBase::Update(const FunctionCallbackInfo<Value>& args) {
   // Only copy the data if we have to, because it's a string
   if (args[0]->IsString()) {
     StringBytes::InlineDecoder decoder;
+    enum encoding enc = ParseEncoding(env->isolate(), args[1], UTF8);
+
+    if (!StringBytes::IsValidString(args[0].As<String>(), enc)) {
+      return THROW_ERR_INVALID_ARG_VALUE(env, "Invalid encoding for data");
+    }
+
```

Comment:
This seems like an anti-pattern to me. You are effectively moving error handling from `Decode()` to different places, however, `ParseEncoding` is now called in multiple different places, too.

Reply:
We prefer [throwing the JS exception in the binding instead of in nested C++ functions](https://github.com/nodejs/node/blob/master/CPP_STYLE_GUIDE.md#avoid-throwing-javascript-errors-in-nested-c-methods).

Ideally this should be done in JS land, though.

Reply:
Moving error handling out is okay as long as the callee still properly checks parameters. Your comment below would mitigate the problem :)

Reply:
I moved the error handling to the binding.